### PR TITLE
add auth and fix some generated code issue in .net

### PR DESCRIPTION
**Pull request by raych1**
What are the changes?
- fixed example path in tspconfig.yaml
- added auth to typespec as it's required to be declared
- use an approved namespace for csharp emitter to avoid the generated code build error
- rename model name `widget` to `widgetsuite` to avoid the generated code analyzer error

@weshaggard , please review.

### CODE REVIEW

File: specification/contosowidgetmanager/Contoso.WidgetManager/tspconfig.yaml [hunk 1]
```diff
@@ -11,21 +11,24 @@ options:
   "@azure-tools/typespec-autorest":
     azure-resource-provider-folder: "data-plane"
     emitter-output-dir: "{project-root}/.."
-    examples-directory: "examples"
+    examples-directory: "{project-root}/examples"
```
Comment (raych1) [inline specification/contosowidgetmanager/Contoso.WidgetManager/tspconfig.yaml:14]: FYI - [issue#3182](https://github.com/Azure/typespec-azure/issues/3182)

File: specification/contosowidgetmanager/Contoso.WidgetManager/tspconfig.yaml [hunk 2]
```diff
@@ -11,21 +11,24 @@ options:
...
     output-file: "{azure-resource-provider-folder}/{service-name}/{version-status}/{version}/widgets.json"
   "@azure-tools/typespec-python":
     package-dir: "azure-contoso-widgetmanager"
     package-mode: "dataplane"
     package-name: "{package-dir}"
   "@azure-tools/typespec-csharp":
-    package-dir: "Azure.Contoso.WidgetManager"
+    package-dir: "Azure.Storage.Contoso"
```
Comment (raych1) [inline specification/contosowidgetmanager/Contoso.WidgetManager/tspconfig.yaml:21]: Talked to @m-nash and agreed to use an approved namespace to succeed the build.

File: specification/contosowidgetmanager/Contoso.WidgetManager/tspconfig.yaml (hunk 1 above)
Comment (weshaggard) [reply specification/contosowidgetmanager/Contoso.WidgetManager/tspconfig.yaml:14]: I'm OK working around this for now but do we know if we can get some of these to be defaults so we don't have to specify them in all the tspconfig files?

File: specification/contosowidgetmanager/Contoso.WidgetManager/tspconfig.yaml (hunk 2 above)
Comment (weshaggard) [reply specification/contosowidgetmanager/Contoso.WidgetManager/tspconfig.yaml:21]: Should we use Azure.Template here instead? https://github.com/Azure/azure-sdk-tools/blob/main/src/dotnet/Azure.ClientSdk.Analyzers/Azure.ClientSdk.Analyzers/ClientAssemblyNamespaceAnalyzer.cs#L34C14-L34C28

**Review by weshaggard — APPROVED** [review]
Left a few comments but looks good

File: specification/contosowidgetmanager/Contoso.WidgetManager/tspconfig.yaml [hunk 3]
```diff
@@ -28,4 +28,4 @@ options:
       name: "@azure-rest/contoso-widgetmanager-rest"
   "@azure-tools/typespec-java":
     package-dir: "azure-contoso-widgetmanager"
-    namespace: com.azure.contoso.widgetmanager
+    namespace: com.azure.contoso.widgetmanager
```
Comment (mikeharder) [inline specification/contosowidgetmanager/Contoso.WidgetManager/tspconfig.yaml:31]: @raych1: Please do not remove trailing newlines when editing files.

**Comment by mikeharder** [comment]
@raych1, @weshaggard: This PR should not have been merged, since pipeline `specs - typespec - ci` was failing.  Going forward, please do not merge PRs if this pipeline is failing.  I have a fix ready in #24744.

**Comment by weshaggard** [comment]
@mikeharder are we at a point were we should make "typespec - ci" a required check for PR's targeting main at least?

**Comment by mikeharder** [comment]
> @mikeharder Mike Harder FTE are we at a point were we should make "typespec - ci" a required check for PR's targeting main at least?

I would say "yes", with the caveat the pipeline name may change once the auto-detection is enabled.

File: specification/contosowidgetmanager/Contoso.WidgetManager/main.tsp [hunk 4]
```diff
@@ -9,21 +9,34 @@ using TypeSpec.Rest;
...
 
+@useAuth(OAuth2Auth<[ContosoAuthFlow]>)
 @service({
   title: "Contoso Widget Manager",
 })
 @versioned(Contoso.WidgetManager.Versions)
+
 namespace Azure.Contoso.WidgetManager;
 
-#suppress "@azure-tools/typespec-azure-core/documentation-required" "https://github.com/Azure/typespec-azure/issues/3107"
```
Comment (mikeharder) [inline specification/contosowidgetmanager/Contoso.WidgetManager/main.tsp:18]: @raych1: Why did you remove this suppression and add the doc comments?  Did you see this is a bug in `typespec-azure` we are waiting for them to fix?

File: specification/contosowidgetmanager/Contoso.WidgetManager/main.tsp [hunk 5]
```diff
@@ -9,21 +9,34 @@ using TypeSpec.Rest;
 using TypeSpec.Versioning;
 using Azure.Core;
 
+@useAuth(OAuth2Auth<[ContosoAuthFlow]>)
```
Comment (mikeharder) [inline specification/contosowidgetmanager/Contoso.WidgetManager/main.tsp:12]: @raych1: Do you think this is the best form of auth to use in the sample?  @tjprescott recommended this fix instead as a reaction to his new linter rule coming in typespec-next:

```
@useAuth(AadOauth2Auth<["https://contoso.azure.com/.default"]>)
```

https://github.com/Azure/azure-rest-api-specs/pull/24628/files#diff-367a7f65ef36d833ded6fe3ba09a1d2232837b92eb652cb9964e7508f31f3274R12

File: specification/contosowidgetmanager/Contoso.WidgetManager/main.tsp (hunk 4 above)
Comment (tjprescott) [reply specification/contosowidgetmanager/Contoso.WidgetManager/main.tsp:18]: We should definitely not put suppressions into our main example. We should take the doc fix.

Comment (mikeharder) [reply specification/contosowidgetmanager/Contoso.WidgetManager/main.tsp:18]: @markcowl: I believe you recommended the opposite, we should suppress this until https://github.com/Azure/typespec-azure/issues/3107 is fixed.

Comment (tjprescott) [reply specification/contosowidgetmanager/Contoso.WidgetManager/main.tsp:18]: Documenting versioning enums is, at best, annoying (but no more annoying than literally every other document nag linter we have). Normalizing the suppression of linter rules in our flagship example does far more harm than good, IMO.

File: specification/contosowidgetmanager/Contoso.WidgetManager/main.tsp (hunk 5 above)
Comment (mikeharder) [reply specification/contosowidgetmanager/Contoso.WidgetManager/main.tsp:12]: @tjprescott recommended we switch to his auth code when we migrate to typespec-next
